addons/mail/models/mail_link_preview.py

```diff
@@ -33,12 +35,16 @@ def _create_from_message_and_notify(self, message):
             return self
         urls = OrderedSet(html.fromstring(message.body).xpath('//a[not(@data-oe-model)]/@href'))
         link_previews = self.env['mail.link.preview']
+        ignore_pattern = self.env["ir.config_parameter"].sudo().get_param("mail.mail_link_preview_ignore_list", "").split("\n")
+        ignore_pattern_regex = re.compile("|".join(ignore_pattern))
```

Comment:
Somehow this should probably be protected against bad writing, you never know what people will put into the configuration parameter. Is compile going to crash or just skip invalid entries ?

Reply:
Any string should be a valid regexp, so a crash is unlikely. I also got rid of the join and test every line.
I don't see what kind of validation I could write for this, do you have some hints?

Reply:
The most simple example: `"("`
`re.error: missing ), unterminated subpattern at position 0`

That said it's a miss-config from admin, the best would be to prevent saving it, but it seems non-critical if there is a crash in this case.

Catching and doing nothing is also acceptable, but then the issue might never be fixed. Posting a message on admin channel? But how to avoid spam?

Reply:
I added some validation, at the very least, each line of the config should `re.compile` without error.
https://github.com/odoo/odoo/pull/177167/files#diff-63280bc62c381158c8fd8ce92902cd70e6fef7bb2b50c0debea8f2044849a5b7